Mathlib/Algebra/BigOperators/Group/Finset/Basic.lean

```diff
@@ -382,6 +382,20 @@ lemma prod_union_eq_left [DecidableEq α] (hs : ∀ a ∈ s₂, a ∉ s₁ → f
 lemma prod_union_eq_right [DecidableEq α] (hs : ∀ a ∈ s₁, a ∉ s₂ → f a = 1) :
     ∏ a ∈ s₁ ∪ s₂, f a = ∏ a ∈ s₂, f a := by rw [union_comm, prod_union_eq_left hs]
 
+/-- The products of two functions `f g : α → β` over finite sets `s₁ s₂ : Finset α`
+are equal if the functions agree on `s₁ ∩ s₂`, `f = 1` and `g = 1` on the respective
+set differences. -/
+@[to_additive "The sum of two functions `f g : α → β` over finite sets `s₁ s₂ : Finset α`
+are equal if the functions agree on `s₁ ∩ s₂`, `f = 0` and `g = 0` on the respective
+set differences."]
+lemma prod_congr_of_eq_on_inter {α β : Type*} {s₁ s₂ : Finset α} {f g : α → β}
+    [CommMonoid β] [DecidableEq α] (h₁ : ∀ a ∈ s₁, a ∉ s₂ → f a = 1)
+    (h₂ : ∀ a ∈ s₂, a ∉ s₁ → g a = 1) (h : ∀ a ∈ s₁ ∩ s₂, f a = g a) :
```

Comment:
Optional: use `∀ a ∈ s₁, a ∈ s₂ → f a = g a` for `h` and drop `DecidableEq α` or use `∀ a ∈ s₁ \ s₂, f a = 1` for `h₁` and similarly for `h₂`.